security/pkg/nodeagent/cache/secretcache_test.go

```diff
@@ -911,12 +913,11 @@ func TestWorkloadAgentGenerateSecretFromFile(t *testing.T) {
 	ctx := context.Background()
 
 	wgAddedWatch.Add(1) // Watch should be added for cert file.
-	notifyEvent.Add(1)  // Nofify should be called once.
-
+	// notify is called only on rotation - it was accidentally called
```

Comment:
nit: I don't think we need to comment about previous bugs in the comments. Sort of confusing when not looking at a diff

Reply:
Usually when I discover something not obvious - I add a comment. I don't have a good  memory,  and maybe others will not have to waste the time in the same place.